### ISSUE TITLE
endpoint /healthz/ready does not show draining or terminating state

### ISSUE THREAD

**Issue by dvob** [istio/istio#32703]
**Description**
We run the ingress-gateways as daemonset and have an external loadbalancer (not part of the Kubernetes cluster) in front of the nodes. The loadbalancer calls the /healthz/ready endpoint to check if the ingress-gateways are ready. If we delete a ingress-gateway pod it does not report its draining state.

This way a external loadbalancer has no way to determine that it has to remove a ingress-gateway from its pool before the pod gets restarted and thus it is not possible to gracfully restart/upgrade the ingress-gateways in such an environment.

It seems that there are two problems:
1. pilot-agent caches the envoy state https://github.com/istio/istio/blob/fa47c1a3b97db0874a76b042b117a0a5fa44beb0/pilot/cmd/pilot-agent/status/ready/probe.go#L74
2. envoy does not report its draining state if `/drain_listeners?graceful` is called. I'm not sure if this is intended but this can be circumvented by calling `curl -X POST http://localhost:15000/healthcheck/fail` in the container preStop lifecycle hook. After calling `healthcheck/fail` envoy does properly report its state as draining.

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
The `pod:15021/healthz/ready` endpoint reports if the ingressgateway pod is draining connections (terminating).

**Steps to reproduce the bug**

Steps to reproduce:
```
kind create cluster
istioctl operator init
kubectl apply -n istio-system -f istiooperator.yaml
```

* `istiooperator.yaml`
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: default
spec:
  meshConfig:
    defaultConfig:
      terminationDrainDuration: 20s
  components:
    ingressGateways:
    - name: istio-ingressgateway
      enabled: true
      k8s:
        env:
        overlays:
        - apiVersion: apps/v1
          kind: Deployment
          name: istio-ingressgateway
          patches:
          - path: kind
            value: DaemonSet
          - path: spec.strategy
          - path: spec.template.spec.containers.[name:istio-proxy].ports.[containerPort:8443].hostPort
            value: 443
          - path: spec.template.spec.containers.[name:istio-proxy].ports.[containerPort:15021].hostPort
            value: 15021
```

* Watch /healthz/ready and delete ingressgateway pod
```
watch -n 1 curl -Ss -w "%{http_code}" http://172.18.0.3:15021/healthz/ready
kubectl -n istio-system delete pod -l app=istio-ingressgateway
```
* curl changes directly from HTTP status code `200` to `Failed to connect to 172.18.0.3 port 15021: Connection refused`

**Version**
```
$ istioctl version
client version: 1.10.0-rc.1
control plane version: 1.10.0-rc.1
data plane version: 1.10.0-rc.1 (1 proxies)
$ kubectl version --short
Client Version: v1.21.0
Server Version: v1.20.2
```

**How was Istio installed?**
Operator

**Comment by dvob** [comment]
Remove the caching (https://github.com/dvob/istio/commit/1b745cbe89346814d53256226b2601933e25cc76) and use the preStop lifecycle hook as follows would work:
```yaml
          - path: spec.template.spec.containers.[name:istio-proxy].lifecycle
            value: {"preStop": {"exec": {"command": ["sh", "-c", "curl -X POST http://localhost:15000/healthcheck/fail"]}}}
```

But probably Envoy should also return its state as DRAINING if `/drain_listeners?graceful` is called and not only if `/healthcheck/fail` is called. At least this is my understandinf from https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/operations/draining

**Comment by ramaraochavali** [comment]
I agree it is better to call that endpoint to show the correct status.  Fixed it in #32892
